Provider AWS crashes with CrashLoopBackOff

<!--
Thank you for helping to improve Crossplane!

Please be sure to search for open issues before raising a new one. We use issues
for bug reports and feature requests. Please find us at https://slack.crossplane.io
for questions, support, and discussion.
-->

### What happened?
The provider-aws controller crashes after creating resources with the master image


### How can we reproduce it?
Error 

`6509560845929043e+09  DEBUG   provider-aws    Reconciling     {"controller": "managed/dbparametergroup.rds.aws.crossplane.io", "request": "/pg12-param-i503158-routes1"}
1.6509560845939696e+09  DEBUG   provider-aws    Reconciling     {"controller": "managed/dbparametergroup.rds.aws.crossplane.io", "request": "/pg13-param-i503158-routes1"}
1.6509560845996935e+09  DEBUG   provider-aws    Reconciling     {"controller": "managed/dbparametergroup.rds.aws.crossplane.io", "request": "/pg96-param-i503158-routes1"}
1.6509560846009943e+09  DEBUG   provider-aws    Reconciling     {"controller": "managed/dbparametergroup.rds.aws.crossplane.io", "request": "/pg10-param-i503158-routes1"}
1.6509560846113737e+09  INFO    controller.managed/userpoolclient.cognitoidentityprovider.aws.crossplane.io     Starting workers        {"reconciler group": "cognitoidentityprovider.aws.crossplane.io", "reconciler kind": "UserPoolClient", "worker count": 10}
1.650956084619773e+09   INFO    controller.providerconfig/providerconfig.aws.crossplane.io      Starting workers        {"reconciler group": "aws.crossplane.io", "reconciler kind": "ProviderConfig", "worker count": 10}
1.6509560846198673e+09  DEBUG   provider-aws    Reconciling     {"controller": "providerconfig/providerconfig.aws.crossplane.io", "request": "/aws-provider-config"}
1.6509560848404026e+09  DEBUG   provider-aws    External resource is up to date {"controller": "managed/cachesubnetgroup.cache.aws.crossplane.io", "request": "/cache-private-i503158-routes1", "uid": "89c09904-8d38-4969-ab22-4ed5ce5f5b28", "version": "54146470", "external-name": "cache-private-i503158-routes1", "requeue-after": 1650956144.8403993}
1.6509560848434489e+09  DEBUG   provider-aws    External resource is up to date {"controller": "managed/internetgateway.ec2.aws.crossplane.io", "request": "/cp-igw-i503158-routes1", "uid": "02a40b3d-dd11-4d28-975f-a5a70356ce5c", "version": "54147805", "external-name": "igw-01bbc9f3d7340de49", "requeue-after": 1650956144.8434463}
1.6509560848490453e+09  DEBUG   provider-aws    External resource is up to date {"controller": "managed/natgateway.ec2.aws.crossplane.io", "request": "/cp-natgw-az3-i503158-routes1", "uid": "f56bf9c4-d21e-45e9-9293-d7cdd6cc5f0e", "version": "54146599", "external-name": "nat-02ac6402d02e245d3", "requeue-after": 1650956144.849043}
1.6509560848500206e+09  DEBUG   provider-aws    External resource is up to date {"controller": "managed/subnet.ec2.aws.crossplane.io", "request": "/cp-pvt-az2-i503158-routes1", "uid": "5f2b4639-67ca-496d-b7de-943b734042c4", "version": "54149796", "external-name": "subnet-0911c24407a255693", "requeue-after": 1650956144.8500183}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x302683b]

goroutine 3575 [running]:
github.com/crossplane/provider-aws/pkg/controller/ec2/route.(*external).findRouteByDestination(0xc0018bd0e0, {0x619d910, 0xc0021281e0}, 0xc00060e500)
        /home/runner/work/provider-aws/provider-aws/pkg/controller/ec2/route/setup.go:119 +0x25b
github.com/crossplane/provider-aws/pkg/controller/ec2/route.(*external).observer(0xc0021ca960, {0x619d910, 0xc0021281e0}, {0x626a220, 0xc00060e500})
        /home/runner/work/provider-aws/provider-aws/pkg/controller/ec2/route/setup.go:80 +0xdb
github.com/crossplane/provider-aws/pkg/controller/ec2/route.(*external).Observe(0xc000ab7710, {0x619d910, 0xc0021281e0}, {0x626a220, 0xc00060e500})
        /home/runner/work/provider-aws/provider-aws/pkg/controller/ec2/route/zz_controller.go:66 +0x6e
github.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*Reconciler).Reconcile(0xc000ac18c0, {0x619d948, 0xc00211c840}, {{{0x0, 0x55787a0}, {0xc001995840, 0x30}}})
        /home/runner/work/provider-aws/provider-aws/vendor/github.com/crossplane/crossplane-runtime/pkg/reconciler/managed/reconciler.go:767 +0x230b
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0xc000ac1970, {0x619d948, 0xc00211c780}, {{{0x0, 0x55787a0}, {0xc001995840, 0x413a94}}})
        /home/runner/work/provider-aws/provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:114 +0x26f
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000ac1970, {0x619d8a0, 0xc00090c700}, {0x50e4660, 0xc0017e95c0})
        /home/runner/work/provider-aws/provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:311 +0x33e
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000ac1970, {0x619d8a0, 0xc00090c700})
        /home/runner/work/provider-aws/provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:266 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
        /home/runner/work/provider-aws/provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:227 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
        /home/runner/work/provider-aws/provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:223 +0x357`

`k get managed
NAME                                                                     VPCID                   READY   SYNCED   AGE
cachesubnetgroup.cache.aws.crossplane.io/cache-private-i503158-routes1   vpc-088ba5f248c9623b2   True    True     18m

NAME                                                           READY   SYNCED   ID                      VPC                     AGE
internetgateway.ec2.aws.crossplane.io/cp-igw-i503158-routes1   True    True     igw-01bbc9f3d7340de49   vpc-088ba5f248c9623b2   18m

NAME                                                       READY   SYNCED   ID                           IP             AGE
address.ec2.aws.crossplane.io/cp-eip-az1-i503158-routes1   True    True     eipalloc-0189229d6abb9b1ae   3.64.30.121    18m
address.ec2.aws.crossplane.io/cp-eip-az2-i503158-routes1   True    True     eipalloc-0b38df36f50cfb726   3.72.117.221   18m
address.ec2.aws.crossplane.io/cp-eip-az3-i503158-routes1   True    True     eipalloc-0ced81df948b5c139   3.73.11.143    18m

NAME                                                              READY   SYNCED   ID                      VPC                     AGE
routetable.ec2.aws.crossplane.io/cp-pub-rtb-i503158-routes1       True    True     rtb-076eff77b57ef1c08   vpc-088ba5f248c9623b2   18m
routetable.ec2.aws.crossplane.io/cp-pvt-rtb-az1-i503158-routes1   False   True     rtb-028a0caef648a4c99   vpc-088ba5f248c9623b2   18m
routetable.ec2.aws.crossplane.io/cp-pvt-rtb-az2-i503158-routes1   True    False    rtb-065b4bf066bc428c8   vpc-088ba5f248c9623b2   18m
routetable.ec2.aws.crossplane.io/cp-pvt-rtb-az3-i503158-routes1   True    False    rtb-071e4be03bd470b9d   vpc-088ba5f248c9623b2   18m

NAME                                               READY   SYNCED   ID                      CIDR          IPV6CIDR   AGE
vpc.ec2.aws.crossplane.io/cp-vpc-i503158-routes1   True    True     vpc-088ba5f248c9623b2   10.0.0.0/16              18m

NAME                                                            READY   SYNCED   ID                      VPC                     SUBNET                     ALLOCATION ID                AGE
natgateway.ec2.aws.crossplane.io/cp-natgw-az1-i503158-routes1   True    True     nat-03963b14854909616   vpc-088ba5f248c9623b2   subnet-002fc4c3ea81b7e0a   eipalloc-0189229d6abb9b1ae   18m
natgateway.ec2.aws.crossplane.io/cp-natgw-az2-i503158-routes1   True    True     nat-035c6ee1b100ba647   vpc-088ba5f248c9623b2   subnet-0204fb176bb4fd28c   eipalloc-0b38df36f50cfb726   18m
natgateway.ec2.aws.crossplane.io/cp-natgw-az3-i503158-routes1   True    True     nat-02ac6402d02e245d3   vpc-088ba5f248c9623b2   subnet-0de8d6d50fd8477f9   eipalloc-0ced81df948b5c139   18m

NAME                                                      READY   SYNCED   ID                         VPC                     CIDR            AGE
subnet.ec2.aws.crossplane.io/cp-pub-az1-i503158-routes1   True    True     subnet-002fc4c3ea81b7e0a   vpc-088ba5f248c9623b2   10.0.101.0/24   18m
subnet.ec2.aws.crossplane.io/cp-pub-az2-i503158-routes1   True    True     subnet-0204fb176bb4fd28c   vpc-088ba5f248c9623b2   10.0.102.0/24   18m
subnet.ec2.aws.crossplane.io/cp-pub-az3-i503158-routes1   True    True     subnet-0de8d6d50fd8477f9   vpc-088ba5f248c9623b2   10.0.103.0/24   18m
subnet.ec2.aws.crossplane.io/cp-pvt-az1-i503158-routes1   True    True     subnet-0aeca82edddfc7f1b   vpc-088ba5f248c9623b2   10.0.1.0/24     18m
subnet.ec2.aws.crossplane.io/cp-pvt-az2-i503158-routes1   True    True     subnet-0911c24407a255693   vpc-088ba5f248c9623b2   10.0.2.0/24     18m
subnet.ec2.aws.crossplane.io/cp-pvt-az3-i503158-routes1   True    True     subnet-0484ba1549278bfc4   vpc-088ba5f248c9623b2   10.0.3.0/24     18m

NAME                                                                        READY   SYNCED   EXTERNAL-NAME
vpcpeeringconnection.ec2.aws.crossplane.io/cp-vpc-peering-i503158-routes1   True    True     pcx-03ee833e1f9b63950

NAME                                                                READY   SYNCED   EXTERNAL-NAME
route.ec2.aws.crossplane.io/cp-pub-rt-igw-i503158-routes1                   False    cp-pub-rt-igw-i503158-routes1
route.ec2.aws.crossplane.io/cp-pub-rt-peering-i503158-routes1               False    cp-pub-rt-peering-i503158-routes1
route.ec2.aws.crossplane.io/cp-pvt-rt-az1-natgw-i503158-routes1     False   False    cp-pvt-rt-az1-natgw-i503158-routes1
route.ec2.aws.crossplane.io/cp-pvt-rt-az1-peering-i503158-routes1           False    cp-pvt-rt-az1-peering-i503158-routes1
route.ec2.aws.crossplane.io/cp-pvt-rt-az2-natgw-i503158-routes1     False   False    cp-pvt-rt-az2-natgw-i503158-routes1
route.ec2.aws.crossplane.io/cp-pvt-rt-az2-peering-i503158-routes1   False   False    cp-pvt-rt-az2-peering-i503158-routes1
route.ec2.aws.crossplane.io/cp-pvt-rt-az3-natgw-i503158-routes1             False    cp-pvt-rt-az3-natgw-i503158-routes1
route.ec2.aws.crossplane.io/cp-pvt-rt-az3-peering-i503158-routes1   False   False    cp-pvt-rt-az3-peering-i503158-routes1

NAME                                                                READY   SYNCED   EXTERNAL-NAME
dbparametergroup.rds.aws.crossplane.io/pg10-param-i503158-routes1   True    True     pg10-param-i503158-routes1
dbparametergroup.rds.aws.crossplane.io/pg11-param-i503158-routes1   True    True     pg11-param-i503158-routes1
dbparametergroup.rds.aws.crossplane.io/pg12-param-i503158-routes1   True    True     pg12-param-i503158-routes1
dbparametergroup.rds.aws.crossplane.io/pg13-param-i503158-routes1   True    True     pg13-param-i503158-routes1
dbparametergroup.rds.aws.crossplane.io/pg96-param-i503158-routes1   True    True     pg96-param-i503158-routes1

NAME                                                                  READY   SYNCED   AGE
dbsubnetgroup.database.aws.crossplane.io/db-private-i503158-routes1   True    True     18m
dbsubnetgroup.database.aws.crossplane.io/db-public-i503158-routes1    True    True     18m`


### What environment did it happen in?
Crossplane version:  crossplane-1.7.0

Include at least the version or commit of Crossplane you were running. Consider
also including your:

* Cloud provider or hardware configuration: AWS
* Kubernetes version
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.3", GitCommit:"816c97ab8cff8a1c72eccca1026f7820e93e0d25", GitTreeState:"clean", BuildDate:"2022-01-25T21:17:57Z", GoVersion:"go1.17.6", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.7", GitCommit:"b56e432f2191419647a6a13b9f5867801850f969", GitTreeState:"clean", BuildDate:"2022-02-16T11:43:55Z", GoVersion:"go1.16.14", Compiler:"gc", Platform:"linux/amd64"}

* Kubernetes distribution: Gardener
* OS (e.g. from /etc/os-release)
* Kernel (e.g. `uname -a`)

--- a/pkg/controller/ec2/route/setup.go
+++ b/pkg/controller/ec2/route/setup.go
@@ -116,7 +116,7 @@ func (e *external) findRouteByDestination(ctx context.Context, cr *svcapitypes.R
 
 	for _, route := range response.RouteTables[0].Routes {
 		if awsclients.StringValue(route.Origin) == svcsdk.RouteOriginCreateRoute {
-			if awsclients.CIDRBlocksEqual(awsclients.StringValue(route.DestinationCidrBlock), *cr.Spec.ForProvider.DestinationCIDRBlock) {
+			if awsclients.CIDRBlocksEqual(awsclients.StringValue(route.DestinationCidrBlock), awsclients.StringValue(cr.Spec.ForProvider.DestinationCIDRBlock)) {
 				return route, nil
 			}
 		}

